app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt

```diff
@@ -236,3 +186,50 @@ class BrowserFragment : WebFragment(), CursorEvent {
         }
     }
 }
+
+private class BrowserIWebViewCallback(
+        private val browserFragment: BrowserFragment
+): IWebView.Callback {
+
+    private var fullscreenCallback: IWebView.FullscreenCallback? = null
+
+    override fun onPageStarted(url: String) {}
+    override fun onPageFinished(isSecure: Boolean) {}
+    override fun onProgress(progress: Int) {}
+
+    override fun onURLChanged(url: String) {}
+    override fun onRequest(isTriggeredByUserGesture: Boolean) {}
+
+    override fun countBlockedTracker() {}
+    override fun resetBlockedTrackers() {}
+
+    override fun onBlockingStateChanged(isBlockingEnabled: Boolean) {}
+
+    override fun onLongPress(hitTarget: IWebView.HitTarget) {}
+
+    override fun onEnterFullScreen(callback: IWebView.FullscreenCallback, view: View?) {
+        fullscreenCallback = callback
+        if (view == null) return
+
+        with (browserFragment) {
+            browserContainer.visibility = View.INVISIBLE
```

Comment:
I liked the comment about why we're hiding browserContainer - for the browser UI and web content.